lib/Sema/TypeCheckNameLookup.cpp

```diff
@@ -618,24 +634,29 @@ diagnoseTypoCorrection(TypeChecker &tc, DeclNameLoc loc, ValueDecl *decl) {
                         isa<ConstructorDecl>(decl) ? "initializer" :
                         isa<FuncDecl>(decl) ? "method" :
                         "member");
-
+      
       return tc.diagnose(parentDecl, diag::note_typo_candidate_implicit_member,
-                         decl->getName().str(), kind);
+                         outName, kind);
     }
   }
 
-  return tc.diagnose(decl, diag::note_typo_candidate, decl->getName().str());
+  return tc.diagnose(decl, diag::note_typo_candidate, outName);
 }
 
 void TypeChecker::noteTypoCorrection(DeclName writtenName, DeclNameLoc loc,
                                      const LookupResult::Result &suggestion) {
   auto decl = suggestion.Decl;
-  auto &&diagnostic = diagnoseTypoCorrection(*this, loc, decl);
+  
+  SmallString<10> nameBuffer;
+  StringRef outName;
+  auto &&diagnostic = diagnoseTypoCorrection(*this, writtenName, loc,
+                                             decl, nameBuffer, outName);
 
   DeclName declName = decl->getFullName();
 
-  if (writtenName.getBaseName() != declName.getBaseName())
-    diagnostic.fixItReplace(loc.getBaseNameLoc(), declName.getBaseName().str());
+  if (writtenName.getBaseName() != declName.getBaseName() ||
+      writtenName.getArgumentNames() != declName.getArgumentNames())
```

Comment:
The fixit needs to replace the entire `writtenName` -- is there a better way to replace _everything_ in the `writtenName` with what's in `declName`?